sql/tests: TestRandomSyntaxGeneration failed

sql/tests.TestRandomSyntaxGeneration [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/11682546?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/11682546?buildTab=artifacts#/) on release-22.2 @ [a9d7853176ce1e2a5c6805c7de6d9ec3a7e1d0f5](https://github.com/cockroachdb/cockroach/commits/a9d7853176ce1e2a5c6805c7de6d9ec3a7e1d0f5):


```
    rsg_test.go:824: 1m15s of 5m0s: 282400 executions, 14868 successful
    rsg_test.go:824: 1m20s of 5m0s: 301181 executions, 15768 successful
    rsg_test.go:824: 1m25s of 5m0s: 319592 executions, 16643 successful
    rsg_test.go:824: 1m30s of 5m0s: 337541 executions, 17487 successful
    rsg_test.go:824: 1m35s of 5m0s: 356303 executions, 18421 successful
    rsg_test.go:824: 1m40s of 5m0s: 375718 executions, 19344 successful
    rsg_test.go:824: 1m45s of 5m0s: 394905 executions, 20187 successful
    rsg_test.go:824: 1m50s of 5m0s: 413323 executions, 21036 successful
    rsg_test.go:824: 1m55s of 5m0s: 431306 executions, 21856 successful
    rsg_test.go:824: 2m0s of 5m0s: 449753 executions, 22756 successful
    rsg_test.go:824: 2m5s of 5m0s: 468146 executions, 23632 successful
    rsg_test.go:824: 2m10s of 5m0s: 488209 executions, 24524 successful
    rsg_test.go:824: 2m15s of 5m0s: 507458 executions, 25398 successful
    rsg_test.go:824: 2m20s of 5m0s: 525411 executions, 26243 successful
    rsg_test.go:824: 2m25s of 5m0s: 543939 executions, 27089 successful
    rsg_test.go:824: 2m30s of 5m0s: 562410 executions, 27969 successful
    rsg_test.go:824: 2m35s of 5m0s: 581276 executions, 28852 successful
    rsg_test.go:824: 2m40s of 5m0s: 601068 executions, 29696 successful
    rsg_test.go:824: 2m45s of 5m0s: 620177 executions, 30547 successful
    rsg_test.go:824: 2m50s of 5m0s: 639800 executions, 31401 successful
    rsg_test.go:824: 2m55s of 5m0s: 658327 executions, 32277 successful
    rsg_test.go:824: 3m0s of 5m0s: 677712 executions, 33160 successful
    rsg_test.go:824: 3m5s of 5m0s: 696504 executions, 34026 successful
    rsg_test.go:824: 3m10s of 5m0s: 713688 executions, 34838 successful
    rsg_test.go:824: 3m15s of 5m0s: 732813 executions, 35722 successful
    rsg_test.go:824: 3m20s of 5m0s: 750826 executions, 36578 successful
    rsg_test.go:824: 3m25s of 5m0s: 769785 executions, 37437 successful
    rsg_test.go:824: 3m30s of 5m0s: 789208 executions, 38284 successful
    rsg_test.go:824: 3m35s of 5m0s: 808411 executions, 39191 successful
    rsg_test.go:824: 3m40s of 5m0s: 826007 executions, 40042 successful
    rsg_test.go:824: 3m45s of 5m0s: 844129 executions, 40936 successful
    rsg_test.go:824: 3m50s of 5m0s: 862973 executions, 41854 successful
    rsg_test.go:824: 3m55s of 5m0s: 881949 executions, 42667 successful
    rsg_test.go:824: 4m0s of 5m0s: 900307 executions, 43459 successful
    rsg_test.go:824: 4m5s of 5m0s: 917309 executions, 44237 successful
    rsg_test.go:824: 4m10s of 5m0s: 935062 executions, 45051 successful
    rsg_test.go:824: 4m15s of 5m0s: 953889 executions, 45893 successful
    rsg_test.go:824: 4m20s of 5m0s: 972788 executions, 46668 successful
    rsg_test.go:824: 4m25s of 5m0s: 991828 executions, 47488 successful
    rsg_test.go:824: 4m30s of 5m0s: 1010001 executions, 48300 successful
    rsg_test.go:824: 4m35s of 5m0s: 1028423 executions, 49108 successful
    rsg_test.go:824: 4m40s of 5m0s: 1047319 executions, 49962 successful
    rsg_test.go:824: 4m45s of 5m0s: 1065840 executions, 50854 successful
    rsg_test.go:824: 4m50s of 5m0s: 1084428 executions, 51746 successful
    rsg_test.go:824: 4m55s of 5m0s: 1102443 executions, 52606 successful
    rsg_test.go:860: 1119662 executions, 53383 successful
    rsg_test.go:868: cannot parse output of Format: sql="PREPARE SMALLINT AS CREATE CHANGEFEED WITH OPTIONS ( BUCKET_COUNT = PLACEHOLDER ) AS SELECT * , * FROM FAMILY AS DECIMAL", formattedSQL="PREPARE \"smallint\" AS CREATE CHANGEFEED WITH bucket_count = 'placeholder' AS SELECT *, * FROM \"family\" AS \"decimal\"": at or near "with": syntax error
    rsg_test.go:306: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/d437d2c847dfedbc4972f231c3331c8e/logTestRandomSyntaxGeneration4263227898
--- FAIL: TestRandomSyntaxGeneration (301.29s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #109946 sql/tests: TestRandomSyntaxGeneration failed [DROP DATABASE timeout] [C-test-failure O-robot T-sql-foundations branch-master release-blocker]
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRandomSyntaxGeneration.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-31336

--- a/pkg/ccl/changefeedccl/changefeed_test.go
+++ b/pkg/ccl/changefeedccl/changefeed_test.go
@@ -5452,17 +5452,17 @@ func TestChangefeedDescription(t *testing.T) {
 		{
 			// TODO(#85143): remove schema_change_policy='stop' from this test.
 			create: "CREATE CHANGEFEED INTO $1 WITH updated, envelope = $2, schema_change_policy='stop' AS SELECT a FROM foo WHERE a % 2 = 0",
-			descr:  `CREATE CHANGEFEED INTO '` + redactedSink + `' WITH envelope = 'wrapped', schema_change_policy = 'stop', updated AS SELECT a FROM foo WHERE (a % 2) = 0`,
+			descr:  `CREATE CHANGEFEED INTO '` + redactedSink + `' WITH OPTIONS (envelope = 'wrapped', schema_change_policy = 'stop', updated) AS SELECT a FROM foo WHERE (a % 2) = 0`,
 		},
 		{
 			// TODO(#85143): remove schema_change_policy='stop' from this test.
 			create: "CREATE CHANGEFEED INTO $1 WITH updated, envelope = $2, schema_change_policy='stop' AS SELECT a FROM public.foo AS bar WHERE a % 2 = 0",
-			descr:  `CREATE CHANGEFEED INTO '` + redactedSink + `' WITH envelope = 'wrapped', schema_change_policy = 'stop', updated AS SELECT a FROM public.foo AS bar WHERE (a % 2) = 0`,
+			descr:  `CREATE CHANGEFEED INTO '` + redactedSink + `' WITH OPTIONS (envelope = 'wrapped', schema_change_policy = 'stop', updated) AS SELECT a FROM public.foo AS bar WHERE (a % 2) = 0`,
 		},
 		{
 			// TODO(#85143): remove schema_change_policy='stop' from this test.
 			create: "CREATE CHANGEFEED INTO $1 WITH updated, envelope = $2, schema_change_policy='stop' AS SELECT a FROM foo WHERE status IN ('open', 'closed')",
-			descr:  `CREATE CHANGEFEED INTO '` + redactedSink + `' WITH envelope = 'wrapped', schema_change_policy = 'stop', updated AS SELECT a FROM foo WHERE status IN ('open', 'closed')`,
+			descr:  `CREATE CHANGEFEED INTO '` + redactedSink + `' WITH OPTIONS (envelope = 'wrapped', schema_change_policy = 'stop', updated) AS SELECT a FROM foo WHERE status IN ('open', 'closed')`,
 		},
 	} {
 		t.Run(tc.create, func(t *testing.T) {

--- a/pkg/ccl/changefeedccl/scheduled_changefeed_test.go
+++ b/pkg/ccl/changefeedccl/scheduled_changefeed_test.go
@@ -236,8 +236,8 @@ func TestSerializesScheduledChangefeedExecutionArgs(t *testing.T) {
 			queryArgs: []interface{}{th.env.Now()},
 			es: expectedSchedule{
 				nameRe:         "foo-changefeed",
-				changefeedStmt: "CREATE CHANGEFEED INTO 'webhook-https://0/changefeed?AWS_SECRET_ACCESS_KEY=nevershown' WITH format = 'JSON', initial_scan = 'only', schema_change_policy = 'stop' AS SELECT * FROM d.public.foo",
-				shownStmt:      "CREATE CHANGEFEED INTO 'webhook-https://0/changefeed?AWS_SECRET_ACCESS_KEY=redacted' WITH format = 'JSON', initial_scan = 'only', schema_change_policy = 'stop' AS SELECT * FROM d.public.foo",
+				changefeedStmt: "CREATE CHANGEFEED INTO 'webhook-https://0/changefeed?AWS_SECRET_ACCESS_KEY=nevershown' WITH OPTIONS (format = 'JSON', initial_scan = 'only', schema_change_policy = 'stop') AS SELECT * FROM d.public.foo",
+				shownStmt:      "CREATE CHANGEFEED INTO 'webhook-https://0/changefeed?AWS_SECRET_ACCESS_KEY=redacted' WITH OPTIONS (format = 'JSON', initial_scan = 'only', schema_change_policy = 'stop') AS SELECT * FROM d.public.foo",
 				period:         time.Hour,
 				runsNow:        true,
 			},

--- a/pkg/sql/parser/testdata/changefeed
+++ b/pkg/sql/parser/testdata/changefeed
@@ -102,15 +102,23 @@ CREATE CHANGEFEED AS SELECT * FROM _ WHERE _ > _ -- identifiers removed
 parse
 CREATE CHANGEFEED WITH opt='val' AS SELECT * FROM foo WHERE a  > b
 ----
-CREATE CHANGEFEED WITH opt = 'val' AS SELECT * FROM foo WHERE a > b -- normalized!
-CREATE CHANGEFEED WITH opt = ('val') AS SELECT (*) FROM foo WHERE ((a) > (b)) -- fully parenthesized
-CREATE CHANGEFEED WITH opt = '_' AS SELECT * FROM foo WHERE a > b -- literals removed
-CREATE CHANGEFEED WITH _ = 'val' AS SELECT * FROM _ WHERE _ > _ -- identifiers removed
+CREATE CHANGEFEED WITH OPTIONS (opt = 'val') AS SELECT * FROM foo WHERE a > b -- normalized!
+CREATE CHANGEFEED WITH OPTIONS (opt = ('val')) AS SELECT (*) FROM foo WHERE ((a) > (b)) -- fully parenthesized
+CREATE CHANGEFEED WITH OPTIONS (opt = '_') AS SELECT * FROM foo WHERE a > b -- literals removed
+CREATE CHANGEFEED WITH OPTIONS (_ = 'val') AS SELECT * FROM _ WHERE _ > _ -- identifiers removed
 
 parse
 CREATE CHANGEFEED INTO 'null://' WITH opt='val' AS SELECT * FROM foo WHERE a  > b
 ----
-CREATE CHANGEFEED INTO 'null://' WITH opt = 'val' AS SELECT * FROM foo WHERE a > b -- normalized!
-CREATE CHANGEFEED INTO ('null://') WITH opt = ('val') AS SELECT (*) FROM foo WHERE ((a) > (b)) -- fully parenthesized
-CREATE CHANGEFEED INTO '_' WITH opt = '_' AS SELECT * FROM foo WHERE a > b -- literals removed
-CREATE CHANGEFEED INTO 'null://' WITH _ = 'val' AS SELECT * FROM _ WHERE _ > _ -- identifiers removed
+CREATE CHANGEFEED INTO 'null://' WITH OPTIONS (opt = 'val') AS SELECT * FROM foo WHERE a > b -- normalized!
+CREATE CHANGEFEED INTO ('null://') WITH OPTIONS (opt = ('val')) AS SELECT (*) FROM foo WHERE ((a) > (b)) -- fully parenthesized
+CREATE CHANGEFEED INTO '_' WITH OPTIONS (opt = '_') AS SELECT * FROM foo WHERE a > b -- literals removed
+CREATE CHANGEFEED INTO 'null://' WITH OPTIONS (_ = 'val') AS SELECT * FROM _ WHERE _ > _ -- identifiers removed
+
+parse
+CREATE CHANGEFEED WITH OPTIONS ( BUCKET_COUNT = PLACEHOLDER ) AS SELECT * , * FROM FAMILY AS DECIMAL
+----
+CREATE CHANGEFEED WITH OPTIONS (bucket_count = 'placeholder') AS SELECT *, * FROM "family" AS "decimal" -- normalized!
+CREATE CHANGEFEED WITH OPTIONS (bucket_count = ('placeholder')) AS SELECT (*), (*) FROM "family" AS "decimal" -- fully parenthesized
+CREATE CHANGEFEED WITH OPTIONS (bucket_count = '_') AS SELECT *, * FROM "family" AS "decimal" -- literals removed
+CREATE CHANGEFEED WITH OPTIONS (_ = 'placeholder') AS SELECT *, * FROM _ AS _ -- identifiers removed

--- a/pkg/sql/sem/tree/changefeed.go
+++ b/pkg/sql/sem/tree/changefeed.go
@@ -62,8 +62,9 @@ func (node *CreateChangefeed) formatWithPredicates(ctx *FmtCtx) {
 		ctx.FormatNode(node.SinkURI)
 	}
 	if node.Options != nil {
-		ctx.WriteString(" WITH ")
+		ctx.WriteString(" WITH OPTIONS (")
 		ctx.FormatNode(&node.Options)
+		ctx.WriteString(")")
 	}
 	ctx.WriteString(" AS ")
 	ctx.FormatNode(node.Select)

